extruct/rdfa.py

```diff
@@ -28,6 +29,104 @@
 
 
 class RDFaExtractor(object):
+    
+    # expands namespace to match with returned json (ex: og -> 'http://ogp.me/ns#')
+    def replaceNS(self, prop, html_element, head_element):
+        
+        # context namespaces taken from pyrdfa3
+        # https://github.com/RDFLib/PyRDFa/blob/master/pyRdfa/initialcontext.py
+        context = {
+            'owl'   : 'http://www.w3.org/2002/07/owl#',
+            'gr'    : 'http://purl.org/goodrelations/v1#',
+            'ctag'    : 'http://commontag.org/ns#',
+            'cc'    : 'http://creativecommons.org/ns#',
+            'grddl'   : 'http://www.w3.org/2003/g/data-view#',
+            'rif'   : 'http://www.w3.org/2007/rif#',
+            'sioc'    : 'http://rdfs.org/sioc/ns#',
+            'skos'    : 'http://www.w3.org/2004/02/skos/core#',
+            'xml'   : 'http://www.w3.org/XML/1998/namespace',
+            'rdfs'    : 'http://www.w3.org/2000/01/rdf-schema#',
+            'rev'   : 'http://purl.org/stuff/rev#',
+            'rdfa'    : 'http://www.w3.org/ns/rdfa#',
+            'dc'    : 'http://purl.org/dc/terms/',
+            'foaf'    : 'http://xmlns.com/foaf/0.1/',
+            'void'    : 'http://rdfs.org/ns/void#',
+            'ical'    : 'http://www.w3.org/2002/12/cal/icaltzd#',
+            'vcard'   : 'http://www.w3.org/2006/vcard/ns#',
+            'wdrs'    : 'http://www.w3.org/2007/05/powder-s#',
+            'og'    : 'http://ogp.me/ns#',
+            'wdr'   : 'http://www.w3.org/2007/05/powder#',
+            'rdf'   : 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+            'xhv'   : 'http://www.w3.org/1999/xhtml/vocab#',
+            'xsd'   : 'http://www.w3.org/2001/XMLSchema#',
+            'v'     : 'http://rdf.data-vocabulary.org/#',
+            'skosxl'  : 'http://www.w3.org/2008/05/skos-xl#',
+            'schema'  : 'http://schema.org/',
+        }
+
+        if 'http://' in prop:
+            return prop
+
+        prefix = prop.split(':')[0]
+        
+        match = None
+        if head_element.get('prefix'):
+            match = re.search(prefix + ': [^\s]+', head_element.get('prefix'))
+
+        # if namespace taken from prefix attribute in head tag
+        if match:
+            ns = match.group().split(': ')[1]
+            return ns + prop.split(':')[1]
+
+        # if namespace taken from xmlns attribute in html tag
+        if ('xmlns:' + prefix) in html_element.keys():
+            return html_element.get('xmlns:' + prefix) + prop.split(':')[1]
+
+        # if namespace present in inital context
+        if prefix in context:
+            return context[prefix] + prop.split(':')[1]
+
+        return prop
+    
+    # sorts the rdfa tags in jsonld string
+    def sort(self, json_object, ordered, key):
```

Comment:
Consider changing this sort algorithm by something like: 
```py
    def sort(self, json_object, ordered, key):
        idx_for_value = dict((value, idx) for idx, value in enumerate(ordered))
        json_object[key].sort(key=lambda props: idx_for_value.get(props.get('@value'), len(ordered)))
```

Also probably better to receive `json_object[key]` directly to save one function parameter

Reply:
Okay I'll optimise this.